core/shared/src/main/scala/hedgehog/package.scala

```diff
@@ -31,12 +31,15 @@ package object hedgehog extends ApplicativeSyntax {
   def propertyT: PropertyTOps =
     new PropertyTOps {}
 
-  implicit class Syntax[A](a1: A) {
+  implicit class Syntax[A](private val a1: A) extends AnyVal {
 
     // FIX Is there a way to get this to work with PropertyT and type-inference?
     def ====(a2: A): Result =
       Result.diffNamed("=== Not Equal ===", a1, a2)(_ == _)
 
-}
+    def matchPattern(right: PartialFunction[A, _]): Result =
```

Comment:
The only suggestion/tweak I might be suggested to make is `PartialFunction[A, Result]` (possibly as a separate method).

Reply:
I think that can be already achieved by just
```scala
actual match {
  case Something(1, _) => Result.success
  case _ => Result.failure
}
```
This PR is more about convenience.